### ISSUE TITLE
Need documentation on storage

### ISSUE THREAD

**Issue by Sriram-Ramaswami** [futurepress/epub.js#295]
The epub.js library needs better documentation on offline storage. For e.g we have no info as to how to override the storage with RAM,indexedDB etc since the library ignores anything except true.

We also dont have information on where in the filesystem it places the epub if we choose the file system option

Thanks,
Sriram

**Comment by fchasen** [comment]
Agreed! We have switched to using https://github.com/mozilla/localForage to store content offline for now, so I need to update the documentation. (For instance filesystem isn't used anymore)

**Comment by shinnlu** [comment]
i have same problem.
when i change the code as below:
```javascript
            var Book = ePub({ restore: true,storage: 'indexeddb' });
            Book.open("http://localhost/epub.js/reader/moby-dick/");
browser return:
localForage library not loadedEPUBJS.Storage.checkRequirements @ storage.js:12EPUBJS.Storage @ storage.js:3EPUBJS.Book.fromStorage @ book.js:1118EPUBJS.Book @ book.js:110root.ePub @ base.js:48(anonymous function) @ basic.html:72
book.js:832 Could not load Chapter: http://localhost/epub.js/reader/moby-dick/OPS/cover.xhtml

**Comment by fchasen** [comment]
`localForage library not loaded` in that error is just saying you need to load the localForage js library.
You can load it as follows:
`<script src="https://cdnjs.cloudflare.com/ajax/libs/localforage/1.3.0/localforage.js"></script>`

I am still working on the documentation but you might want to start with the offline.html example as a base: https://github.com/futurepress/epub.js/blob/master/examples/offline.html

**Comment by paulocheque** [comment]
When I set `storage: true` I am getting this error: `TypeError: this.store.isStored is not a function`.

When I set `restore: true` without the `storage` property, I dont receive any errors, but the `online`, `offline` and `stored` events are not being called, so I don't think it is loading from the localForage.

I am using NPM with the following packages:
- epub 0.2.14
- localforage 1.4.2 
- rsvp 3.2.1

Any ideas?
Thanks in advance

**Comment by Kilbourne** [comment]
@paulocheque found any solution?

**Comment by paulocheque** [comment]
@Kilbourne I end up to create my own storage solution. So I use `restore: false` and `storage: false`, then I save in the local storage by myself.

**Comment by Mrjaid** [comment]
@ paulocheque please can you kindly share?? need to me able to highlight a text and change highlight background and to retrieve doc in this state later.

**Comment by paulocheque** [comment]
I did something very simple:

    window.localStorage.getItem(key)
    window.localStorage.setItem(`${key}`, 'something I need to be saved')

You can even use the LocalForage library for that.

**Comment by Mrjaid** [comment]
@paulocheque Many thanks for the feed back, but can you kindly show how I could save a book file and read it from local storage and not from file... do I save the html after display as i want users to be able to highlight text on the book?

**Comment by paulocheque** [comment]
I did not save the book. I just used HTTP cache for that.

But follow a piece of code of the epub.js library that may help you:

```
localStorage.setItem(this.settings.bookKey, JSON.stringify(this.contents));
```
